include/meshio/vectors.hpp

```diff
@@ -96,44 +181,106 @@ class Vec3 {
     }
 
     bool operator==(const Vec3& other) {
-        return ((x == other.x) && (y == other.y) && (z == other.z));
+        return ((elements[0] == other.X()) && (elements[1] == other.Y()) &&
+                (elements[2] == other.Z()));
     }
+
+  private:
+    T elements[3];
 };
 
 template<class T>
-class Vec4 {
+class Vec4 : public VecBase<T> {
   public:
-    T x,y,z,w;
+    typedef T *iterator;
+    typedef const T *const_iterator;
 
-    Vec4(T pX, T pY, T pZ, T pW)
-        : x(pX), y(pY), z(pZ), w(pW) {
+    Vec4(T pX, T pY, T pZ, T pW) : elements{pX, pY, pZ, pW} {}
+    Vec4() : Vec4({static_cast<T>(0), static_cast<T>(0), static_cast<T>(0),
+                  static_cast<T>(0)}) {}
```

Comment:
@9prady9 Not sure why delegating constructor here resulted in compilation error with gcc, but worked with clang... !!!